src/sentry/sentry_apps/token_exchange/refresher.py

```diff
@@ -78,18 +85,36 @@ def token(self) -> ApiToken:
         try:
             return ApiToken.objects.get(refresh_token=self.refresh_token)
         except ApiToken.DoesNotExist:
-            raise APIUnauthorized("Token does not exist")
+            raise SentryAppIntegratorError(
+                message="Given refresh token does not exist",
+                webhook_context={
+                    "token": self.refresh_token[:4],
+                    "installation_uuid": self.install.uuid,
+                },
+            )
 
     @cached_property
     def application(self) -> ApiApplication:
         try:
             return ApiApplication.objects.get(client_id=self.client_id)
         except ApiApplication.DoesNotExist:
-            raise APIUnauthorized("Application does not exist")
+            raise SentryAppIntegratorError(
+                message="Could not find matching Application for given client_id",
+                webhook_context={
+                    "client_id": self.client_id[:4],
+                    "installation_uuid": self.install.uuid,
+                },
+            )
```

Comment:
`APIUnauthorized` has status code 401

```suggestion
            raise SentryAppIntegratorError(
                message="Could not find matching Application for given client_id",
                webhook_context={
                    "client_id": self.client_id[:4],
                    "installation_uuid": self.install.uuid,
                },
                status_code=401
            )
```